pkg/apis/allocation/v1/gameserverallocation.go

```diff
@@ -564,6 +565,13 @@ type GameServerMetadata struct {
 
 // ApplyDefaults applies the default values to this GameServerAllocation
 func (gsa *GameServerAllocation) ApplyDefaults() {
+	if len(gsa.Spec.Priorities) == 0 {
```

Comment:
This implementation isn't correct unfortunately - #3734 has it correct - need to loop around each one.

Reply:
I haven't seen the other PR. Since the correct approach is in the above mentioned PR, I will close this PR.